src/main/java/com/greenfox/tribesoflagopus/backend/service/DtoService.java

```diff
@@ -49,13 +51,27 @@ public UserDto createUserDto(User user) {
   public List<BuildingDto> convertFromBuildings(List<Building> buildings) {
     List<BuildingDto> listOfBuildingDtos = new ArrayList<>();
     for (Building building : buildings) {
-      BuildingDto buildingDto = BuildingDto.builder()
-          .id(building.getId())
-          .type(building.getType())
-          .level(building.getLevel())
-          .hp(building.getHp())
-          .build();
-      listOfBuildingDtos.add(buildingDto);
+      if (building.getType().equals(BuildingType.TOWNHALL)) {
+        BuildingDto buildingDto = BuildingDto.builder()
+            .id(building.getId())
+            .type(building.getType())
+            .level(building.getLevel())
+            .hp(building.getHp())
+            .startedAt(building.getStartedAt())
+            .finishedAt(building.getStartedAt())
```

Comment:
You probably won't need this if-else statement here.
The finishedAt == startedAt statement in case of Townhall will be true only in case of registration, while your townhall is level 1. After that the startedAt and finishedAt values will represent the timestamps of the latest upgrades, which upgrades need time, so these timestamps won't be equal anymore.
You should handle this logic in your `calculateFinishedAtTime()` method, instead of an if-else statement here.